### PR TITLE
Fix: Increase password max length and accept '-' for username in regex

### PR DESCRIPTION
**Pull request by fuegovic**
fix issue: #561 
fix issue: https://discord.com/channels/1086345563026489514/1121274461887402004

Raise the password max length to 128 characters
Add "-" to the regex
Allow pasting in the "password confirmation" box

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings

### CODE REVIEW

File: client/src/components/Auth/Registration.tsx [hunk 1]
```diff
@@ -187,8 +187,8 @@ function Registration() {
                     message: 'Password must be at least 8 characters'
                   },
                   maxLength: {
-                    value: 40,
-                    message: 'Password must be less than 40 characters'
+                    value: 128,
+                    message: 'Password must be less than 128 characters'
```
Comment (Infinoid) [inline client/src/components/Auth/Registration.tsx]: Minor note: if 128 is the max length, then the password must be "less than or equal to", not "less than" 128 characters.

File: client/src/components/Auth/ResetPassword.tsx [hunk 2]
```diff
@@ -94,8 +94,8 @@ function ResetPassword() {
                       message: 'Password must be at least 8 characters'
                     },
                     maxLength: {
-                      value: 40,
-                      message: 'Password must be less than 40 characters'
+                      value: 128,
+                      message: 'Password must be less than 128 characters'
```
Comment (Infinoid) [inline client/src/components/Auth/ResetPassword.tsx]: Minor note: if 128 is the max length, then the password must be "less than or equal to", not "less than" 128 characters.

File: client/src/components/Auth/LoginForm.tsx [hunk 3]
```diff
@@ -74,7 +74,7 @@ function LoginForm({ onSubmit }: TLoginFormProps) {
               },
               maxLength: {
                 value: 40,
-                message: 'Password must be less than 40 characters'
+                message: 'Password must be less than 128 characters'
```
Comment (Infinoid) [inline client/src/components/Auth/LoginForm.tsx]: Minor note: if 128 is the max length, then the password must be "less than or equal to", not "less than" 128 characters.

**Comment by Infinoid** [comment]
I'm not a project developer, but thanks for submitting this.  I took a quick look at it, and commented on some cases where the error message was not accurate.

Looking at it a little more, it looks like there's also some inconsistency in the *minimum* length.  (6 in the api backend, 8 in the client frontend.)  Though I guess a user would never see that unless they're accessing the API directly.

File: client/src/components/Auth/Registration.tsx (hunk 1 above)
Comment (fuegovic) [reply client/src/components/Auth/Registration.tsx]: Thanks for pointing that out

**Comment by fuegovic** [comment]
@Infinoid Thanks for pointing that out, I've set the min password length to 8 and changed the max length message

**Review by danny-avila — APPROVED** [review]

**Pull request merged**